content/en/docs/trustee/key-broker-service/_index.md

```diff
@@ -0,0 +1,69 @@
+---
+title: Key Broker Service (KBS)
+description: This service facilitates remote attestation and secret delivery
+weight: 1
+categories:
+- docs
+tags:
+- docs
+- kbs
+---
+
+The Confidential Containers Key Broker Service (KBS) facilitates remote attestation and secret delivery. The KBS is an implementation of a [Relying Party](https://www.ietf.org/archive/id/draft-ietf-rats-architecture-22.html#name-relying-party) from the [Remote ATtestation ProcedureS (RATS) Architecture](https://www.ietf.org/archive/id/draft-ietf-rats-architecture-22.html). The KBS itself does not validate attestation evidence. Instead, it relies on the [Attestation-Service (AS)](../attestation-service/) to verify TEE evidence.
+
+In conjunction with the AS or Intel Trust Authority (ITA), the KBS supports the following TEEs:
+
+- AMD SEV-SNP
+- AMD SEV-SNP on Azure with vTPM
+- Intel TDX
+- Intel TDX on Azure with vTPM
+- Intel SGX
+- ARM CCA
+- Hygon CSV
+
+## Deployment Configurations
+
+The KBS can be deployed in several different environments, including as part of a docker compose cluster, part of a Kubernetes cluster or without any containerization. Additionally, the KBS can interact with other attestation components in different ways. This section focuses on the different ways the KBS can interact with other components.
+
+### Background Check Mode
+
+Background check mode is the most common way to configure the Key Broker Service (KBS) and Attestation-Service (AS). The term ["Background Check"](https://www.ietf.org/archive/id/draft-ietf-rats-architecture-22.html#section-5.2) is from the RATS architecture. In background check mode, the KBS releases secrets to a confidential guest after the attestation agent has validated the hardware evidence.
```

Comment:
I'm not sure this is accurate anymore. For some time now, in a default CoCo deployment, AA will get a token from a token-kbs and use that token to retrieve secrets from a resource-kbs (token-kbs + resource-kbs can be the same instance, but it's not bg-check any more, the evidence is just verified & appraised once unless the token expires)

Reply:
maybe we could just say `more straightforward and simple` rather than `the most common` to avoid uncertained indication.

```suggestion
Background check mode is a more straightforward and simple way to configure the Key Broker Service (KBS) and Attestation-Service (AS). The term ["Background Check"](https://www.ietf.org/archive/id/draft-ietf-rats-architecture-22.html#section-5.2) is from the RATS architecture. In background check mode, the KBS directly forwards the hardware evidence of a confidential guest to the AS to validate. Once the validation passes, the KBS will release secrets to the confidential guest.
```

Reply:
> I would still call the default deployment a background check model, though.

hmm, but the verification is only performed once per token lifetime, calling it a bg-check model in which every key release implies a discrete evidence verification would be confusing, no? e.g. if you change the reference values after someone successfully acquired the token, a resource-kbs will still hand out keys, even if the TEE doesn't qualify any more.

Reply:
I think regardless of which model you use the AS policy is run once per token and the KBS policy is run for each resource request. Our delineation between bgcheck and passport is pretty fuzzy especially since we still need to KBS for passport mode. I think the clearest way to talk about it is that bgcheck = 1 KBS and passport mode = 2. That isn't quite the rats definitions but we don't quite follow the rats model here.